docs/book/src/developer/testing.md

```diff
@@ -53,36 +53,100 @@ The following guidelines should be followed when developing E2E tests:
 
 See [e2e development] for more information on developing e2e tests for CAPI and external providers.
 
-## Running the end-to-end tests
+## Running the end-to-end tests locally
 
-`make docker-build-e2e` will build the images for all providers that will be needed for the e2e test.
+Usually, the e2e tests are executed by Prow either pre-submit (on PRs) or periodically on certain branches
+(e.g. the default branch). Those jobs are defined in the kubernetes/test-infra repository in [config/jobs/kubernetes-sigs/cluster-api](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/cluster-api).
+For development and debugging those tests can also be executed locally.
+
+### Prerequisites
+
```

Comment:
does the user need anything beyond docker and the make/build commands?

Reply:
I think it depends on which exact test is executed and in which way it is executed. Afaik:
* make
  * requires at least kustomize and kind (ci-e2e.sh installs them via hack/ensure-*.sh)
* scripts/ci-e2e.sh
  *  when new kindest/node images have to be build (i.e. they don't exist remotely) it builds the images locally and depends on the existence of "$GOPATH/src/k8s.io/kubernetes"
* IDE: same as scripts/ci-e2e.sh, because it's used to build the kindest/node images

Looking at `ci-e2e.sh` we make sure to have the following installed: go, kubectl, kustomize, kind.

I can document this on a best effort basis. Maybe @fabriziopandini knows some additional dependencies.

Reply:
Let's start with what we have now, and then we can eventually iterate adding more context.
However, generally speaking we should document in single place the dependencies for a developer workstation, and reference this list whenever necessary, like in this page.
As of today we already have something similar in two places, which is not ideal
- https://master.cluster-api.sigs.k8s.io/developer/guide.html
- https://master.cluster-api.sigs.k8s.io/developer/tilt.html

So I think we should defer fixing this to a follow up PR

Reply:
> I think it depends on which exact test is executed and in which way it is executed. 

@sbueringer, i was just thinking about the bare minimum, not necessarily all the options

> As of today we already have something similar in two places, which is not ideal

++, would be nice to have a single place that we could link